pkg/osutil/osutil.go

```diff
@@ -149,11 +151,79 @@ func replaceTemplateVariables(fileSys fs.FS, srcPath string, draftConfig *config
 	}
 
 	fileString := string(file)
-
 	for _, variable := range draftConfig.Variables {
 		log.Debugf("replacing %s with %s", variable.Name, variable.Value)
 		fileString = strings.ReplaceAll(fileString, "{{"+variable.Name+"}}", variable.Value)
 	}
-
 	return []byte(fileString), nil
 }
+
+// CopyDirWithTemplates - Handles Gotemplate processing and writing
+func CopyDirWithTemplates(
+	fileSys fs.FS,
+	src, dest string,
+	draftConfig *config.DraftConfig,
+	templateWriter templatewriter.TemplateWriter) error {
+
+	files, err := fs.ReadDir(fileSys, src)
+	if err != nil {
+		return err
+	}
+
+	for _, f := range files {
+
+		if f.Name() == "draft.yaml" {
+			continue
+		}
+
+		srcPath := path.Join(src, f.Name())
```

Comment:
bring in the filename handling from a recent PR: https://github.com/Azure/draft/commit/66436cab57596b6d51d2708e5b86d0237e8e73b7#diff-126f3c7e552bc0e8c372842ae459a986c738fedb8cb628a115b43c8cc19010c5

Should behave the same as the existing CopyDir and this snuck in during your changes